azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/PackageMojo.java

```diff
@@ -86,12 +87,22 @@ protected AnnotationHandler getAnnotationHandler() throws Exception {
     protected Set<Method> findAnnotatedMethods(final AnnotationHandler handler) throws Exception {
         info("");
         info(SEARCH_FUNCTIONS);
-        final Set<Method> functions = handler.findFunctions(getClassUrl());
+        Set<Method> functions;
+        try {
+            functions = handler.findFunctions(getTargetClassUrl());
+        } catch (NoClassDefFoundError e) {
+            // fallback to reflect through artifact url
+            functions = handler.findFunctions(getArtifactUrl());
+        }
         info(functions.size() + FOUND_FUNCTIONS);
         return functions;
     }
 
-    protected URL getClassUrl() throws Exception {
+    protected URL getArtifactUrl() throws Exception {
+        return this.getProject().getArtifact().getFile().toURI().toURL();
```

Comment:
Also let's add a new integration test case to prevent regression.

Reply:
ok